include/deal.II/lac/sparse_matrix_tools.h

```diff
@@ -166,7 +167,12 @@ namespace SparseMatrixTools
                        comm);
           AssertThrowMPI(ierr);
 
-          T sum = Utilities::MPI::sum(value, comm);
+          // Then we also need the total sum. We could obtain it by
+          // calling Utilities::MPI::sum(), but it is cheaper by about
+          // a factor of 2 if we broadcast it from the last process, which
+          // can compute it from its own prefix sum plus its own value.
```

Comment:
It is usually not as bad as a factor of 2 at least for small sizes, because `Allreduce` typically only costs one single communication step more than the `log(n)` communications of `Reduce` or `Bcast` when seen from the most expensive rank. The algorithm is called recursive doubling and it makes all processes participate during the reduction, i.e., it does not try to minimize the number of communications over the whole machine, but the length of the dependency chain.

The change is still correct, but I would simply write that "it is cheaper if we broadcast from the last process".

Reply:
Fair enough. I've just removed the "by about a factor of 2" :-)